tests/Message/ExtendedFetchTransactionRequestTest.php

```diff
@@ -215,4 +217,117 @@ public function testSendFailure()
+            'DATE_CREATED' => $date_created,
+            'CURRENCY' => $currency,
+            'CUSTOMER_REFERENCE' => $customer_reference,
+            'TRANSACTION_REFERENCE' => $this->transactionReference,
+            'STATUS' => $status,
+        ));
+
+        $response = $this->request->send();
+        $transaction = $response->getTransaction();
+        $this->assertSame($amount, $transaction->getAmount());
+        $this->assertSame($currency, $transaction->getCurrency());
+        $this->assertSame($status, $transaction->getStatus());
+        $this->assertSame($customer_reference, $transaction->getCustomerReference());
+        $this->assertSame($date_created, $transaction->getDate()->format('d-M-y'));
+        $this->assertSame($this->transactionReference, $transaction->getTransactionReference());
+    }
+
+    /**
+     * @return void
+     * @psalm-suppress PossiblyNullArrayAccess
+     */
+    public function testGetRefunds()
+    {
+        $currency = $this->faker->currency();
+        $amount = $this->faker->monetaryAmount($currency);
+        $customer_reference = $this->faker->customerReference();
+        $date_created = $this->faker->timestamp();
+        $reversal_reference = $this->faker->transactionReference();
+        $reversal_date = $this->faker->timestamp();
+
+        $this->setMockHttpResponse('ExtendedFetchTransactionWithReversalSuccess.txt', array(
+            'AMOUNT' => $amount,
+            'CHARGE_DATE' => $date_created,
+            'CURRENCY' => $currency,
+            'CUSTOMER_REFERENCE' => $customer_reference,
+            'REVERSAL_REFERENCE' => $reversal_reference,
+            'REVERSAL_AMOUNT' => $amount,
+            'REVERSAL_DATE' => $reversal_date,
+            'REVERSAL_TYPE' => Constants::REVERSAL_REFUND,
+            'TRANSACTION_REFERENCE' => $this->transactionReference,
+        ));
+
+        $response = $this->request->send();
+
+        $refunds = $response->getRefunds();
+        $this->assertNotNull($refunds);
+        $this->assertNull($response->getChargebacks());
+        $this->assertCount(1, $refunds);
+
+        /** @var Refund $refund */
+        $refund = $refunds[0];
+        $this->assertSame($amount, $refund->getAmount());
+        $this->assertSame($currency, $refund->getCurrency());
+        $this->assertSame($reversal_reference, $refund->getRefundReference());
+        $this->assertSame($reversal_date, $refund->getTime());
+        $this->assertSame($this->transactionReference, $refund->getTransactionReference());
+    }
+
+    /**
+     * @return void
+     * @psalm-suppress PossiblyNullArrayAccess
+     */
+    public function testGetChargebacks()
+    {
+        $currency = $this->faker->currency();
+        $amount = $this->faker->monetaryAmount($currency);
+        $customer_reference = $this->faker->customerReference();
+        $date_created = $this->faker->timestamp();
+        $reversal_reference = $this->faker->transactionReference();
```

Comment:
is "reversal" BlueSnap terminology?

Reply:
It's my first time hearing it used in this context but it's also an industry term